peer/example_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2015 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package peer_test
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/btcsuite/btcd/addrmgr"
+	"github.com/btcsuite/btcd/peer"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// zeroHash is the zero value hash (all zeros).  It is defined as a convenience.
+var zeroHash wire.ShaHash
+
+// Not implemented
+func lookupFunc(host string) ([]net.IP, error) {
+	return nil, errors.New("not implemented")
+}
+
+// newestSha returns the latest known block to this peer
+func newestSha() (sha *wire.ShaHash, height int32, err error) {
+	hash := "14a0810ac680a3eb3f82edc878cea25ec41d6b790744e5daeef"
+	height = 234439
+	sha, err = wire.NewShaHashFromStr(hash)
+	return
```

Comment:
This is a good example of why we don't use named returns.  In this case, a height of 234439 would be returned instead of 0 if an error were to occur.

Also, we've been trying to work towards removing variables named `sha` from the code base since the hash might not always be a sha if the hashing function changes.

I would suggest the following:

```Go
// newestSha returns the latest known block to this peer.  It returns a
// hard-coded hash and height since this is only used for example code.
func newestSha() (*wire.ShaHash, int32, error) {
	hashStr := "14a0810ac680a3eb3f82edc878cea25ec41d6b790744e5daeef"
	hash, err := wire.NewShaHashFromStr(hashStr)
	if err != nil {
		return nil, 0, err
	}
	return hash, 234439, nil
}
```